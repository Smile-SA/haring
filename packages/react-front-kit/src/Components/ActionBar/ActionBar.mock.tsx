import type { IActionBarProps } from './ActionBar';
import type { IThumbnail, IThumbnailAction } from '../../types';

import { Trash } from '@phosphor-icons/react';
import { FolderMove } from '@smile/react-front-kit-shared';
import { action } from '@storybook/addon-actions';

export const actionBarActionsMock: IThumbnailAction[] = [
  {
    icon: <FolderMove size={16} />,
    id: 'move',
    isItemAction: false,
    isMassAction: true,
    label: 'Move in tree',
    onAction: action('Move selected in tree'),
  },
  {
    color: 'red',
    confirmModalProps: {
      cancelLabel: 'Abort',
      children: <p>Are you sure ?</p>,
      confirmColor: 'red',
      confirmLabel: 'Remove',
      title: 'Remove files ?',
    },
    confirmation: true,
    icon: <Trash size={16} />,
    id: 'delete',
    isItemAction: false,
    isMassAction: true,
    label: 'Delete',
    onAction: action('Delete selected'),
  },
];

export const actionBarSelectedMock: IThumbnail[] = [
  {
    id: '1',
    label: 'Label A',
  },
  {
    id: '2',
    label: 'Label B',
  },
  {
    id: '3',
    label: 'Label C',
  },
];

export const actionBarLabelMock = (elements: number): string =>
  `${elements} selected`;

export const actionBarMock: IActionBarProps<IThumbnail> = {
  actions: actionBarActionsMock,
  selectedElements: actionBarSelectedMock,
  selectedElementsLabel: actionBarLabelMock,
};
