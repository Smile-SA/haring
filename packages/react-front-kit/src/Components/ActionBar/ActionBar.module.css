.actionBar {
  align-items: center;
  background-color: var(--mantine-primary-color-9);
  border-radius: 4px;
  /*color: theme.white;*/
  display: inline-flex;
  justify-content: space-between;
  padding: 16px 24px;
  /*ref: getStylesRef(actionBar);*/
  width: 100%;
}
