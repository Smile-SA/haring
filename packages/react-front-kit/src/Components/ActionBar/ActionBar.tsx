'use client';

import type { GroupProps, ModalProps } from '@mantine/core';
import type { Record } from '@phosphor-icons/react';
import type {
  IAction,
  IActionConfirmModalProps,
} from '@smile/react-front-kit-shared';
import type { ReactElement } from 'react';

import { Button, Group } from '@mantine/core';
import { createStyles } from '@mantine/styles';
import { useState } from 'react';

import { ConfirmModal } from '../ConfirmModal/ConfirmModal';

const useStyles = createStyles((theme) => ({
  actionBar: {
    alignItems: 'center',
    background: theme.fn.primaryColor(),
    borderRadius: 4,
    color: 'white',
    display: 'inline-flex',
    justifyContent: 'space-between',
    padding: '16px 24px',
    width: '100%',
  },
}));

export type IActionBarAction<Data extends Record<string, unknown>> = IAction<
  Data[]
>;

export interface IActionBarProps<Data extends Record<string, unknown>>
  extends GroupProps {
  actions?: IActionBarAction<Data>[];
  modalProps?: Omit<ModalProps, 'title'>;
  selectedElements: Data[];
  selectedElementsLabel?: (selectedElements: number) => string;
}

export function ActionBar<Data extends Record<string, unknown>>(
  props: IActionBarProps<Data>,
): ReactElement {
  const {
    actions,
    modalProps,
    selectedElements,
    selectedElementsLabel = (selectedElements: number) =>
      `${selectedElements} file(s) selected`,
    ...groupProps
  } = props;
  const [confirmAction, setConfirmAction] = useState<IActionConfirmModalProps<
    Data | Data[]
  > | null>(null);
  const numberOfSelectedElements = selectedElements.length;

  const { classes } = useStyles();

  function setModal(action: IActionBarAction<Data>): void {
    setConfirmAction({
      cancelColor: action.confirmModalProps?.cancelColor,
      cancelLabel: action.confirmModalProps?.cancelLabel,
      children: action.confirmModalProps?.children,
      confirmColor: action.confirmModalProps?.confirmColor,
      confirmLabel: action.confirmModalProps?.confirmLabel,
      onConfirm: () => action.onAction?.(selectedElements),
      title: action.confirmModalProps?.title,
    });
  }

  function clearConfirmAction(): void {
    setConfirmAction(null);
  }

  function handleClose(): void {
    clearConfirmAction();
  }

  function handleModalButton(onAction?: (item: Data[]) => void): void {
    onAction?.(selectedElements);
    handleClose();
  }

  function handleGridAction(action: IActionBarAction<Data>): void {
    if (action.confirmation) {
      setModal(action);
    } else {
      action.onAction?.(selectedElements);
    }
  }

  return (
    <>
      <div className={classes.actionBar}>
        <span>{selectedElementsLabel(numberOfSelectedElements)}</span>
        {actions && actions.length > 0 ? (
          <Group {...groupProps}>
            {actions.map((action) => (
              <Button
                key={action.id}
                color={action.color}
                leftIcon={
                  typeof action.icon === 'function'
                    ? action.icon(selectedElements)
                    : action.icon
                }
                onClick={() => handleGridAction(action)}
                variant={action.color ? 'filled' : 'default'}
              >
                {typeof action.label === 'function'
                  ? action.label(selectedElements)
                  : action.label}
              </Button>
            ))}
          </Group>
        ) : null}
      </div>
      <ConfirmModal
        {...confirmAction}
        onCancel={() => handleModalButton(confirmAction?.onCancel)}
        onClose={handleClose}
        onConfirm={() => handleModalButton(confirmAction?.onConfirm)}
        opened={Boolean(confirmAction)}
        {...modalProps}
      >
        {confirmAction?.children}
      </ConfirmModal>
    </>
  );
}
