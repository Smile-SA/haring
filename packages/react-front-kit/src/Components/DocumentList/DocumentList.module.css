.list {
  & :global(.actionBarRef) {
    margin-bottom: 1.5rem;
  }

  & :global(.selectableListRef) {
    margin-top: 0 !important;
  }

  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    & :global(.selectableListCheckboxRef) {
      position: absolute;
      top: 12.5px;
    }

    & :global(.documentBoxIconRef) {
      left: 32px;
    }

    & :global(.documentBoxHeaderTopRef) {
      margin-left: 72px;
    }
  }
}
