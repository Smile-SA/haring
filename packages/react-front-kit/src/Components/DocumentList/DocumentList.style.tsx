import { createStyles, getStylesRef } from '@mantine/styles';

export const useStyles = createStyles((theme) => ({
  list: {
    [`.${getStylesRef('actionBar')}`]: {
      marginBottom: '1.5rem',
    },
    [`.${getStylesRef('selectableList')}`]: {
      marginTop: 0,
    },
    [theme.fn.smallerThan('sm')]: {
      [`.${getStylesRef('selectableListCheckbox')}`]: {
        position: 'absolute',
        top: 12.5,
      },
      [`.${getStylesRef('documentCardIcon')}`]: {
        left: 32,
      },
      [`.${getStylesRef('documentCardHeaderTop')}`]: {
        marginLeft: 72,
      },
    },
  },
}));
