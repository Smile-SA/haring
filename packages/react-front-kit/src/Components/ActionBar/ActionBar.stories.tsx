import type { IActionBarAction } from './ActionBar';
import type { Meta, StoryObj } from '@storybook/react';

import { ActionBar as Cmp } from './ActionBar';
import {
  actionBarActionsMock,
  actionBarLabelMock,
  actionBarSelectedMock,
} from './ActionBar.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/ActionBar',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const ActionBar: IStory = {
  args: {
    actions: actionBarActionsMock as IActionBarAction<
      Record<string, unknown>
    >[],
    selectedElements: actionBarSelectedMock,
    selectedElementsLabel: actionBarLabelMock,
  },
};
