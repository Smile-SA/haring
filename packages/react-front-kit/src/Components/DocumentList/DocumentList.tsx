import type { IActionBarAction, IActionBarProps } from '../ActionBar/ActionBar';
import type { ISelectableListProps } from '../SelectableList/SelectableList';
import type { ReactElement, ReactNode } from 'react';

import { isNotNullNorEmpty } from '@smile/react-front-kit-shared';
import { useMemo } from 'react';

import { ActionBar } from '../ActionBar/ActionBar';
import { DocumentCard } from '../DocumentCard/DocumentCard';
import { SelectableList } from '../SelectableList/SelectableList';

export interface IDocument extends Record<string, unknown> {
  author?: string;
  content?: ReactNode;
  date?: string;
  iconType?: string;
  id: number | string;
  path?: string;
  title?: string;
}

export interface IDocumentListProps
  extends Omit<
    ISelectableListProps,
    'children' | 'onSelectChange' | 'selectedIndexes'
  > {
  actionBarProps?: Omit<
    IActionBarProps<IDocument>,
    'actions' | 'selectedElements'
  >;
  actions?: IActionBarAction<IDocument>[];
  documents: IDocument[];
  onDocumentSelected?: (document: IDocument, isSelected: boolean) => void;
  selectedDocuments: IDocument[];
}

export function DocumentList(props: IDocumentListProps): ReactElement {
  const {
    actionBarProps,
    actions,
    documents,
    selectedDocuments,
    onDocumentSelected,
    ...selectableListProps
  } = props;
  const selectedIndexes = useMemo(
    () =>
      documents
        .map((document, i) =>
          selectedDocuments.map((selected) => selected.id).includes(document.id)
            ? i
            : null,
        )
        .filter(isNotNullNorEmpty),
    [documents, selectedDocuments],
  );

  function handleSelectChange(index: number, isSelected: boolean): void {
    onDocumentSelected?.(documents[index], isSelected);
  }

  return (
    <div>
      {selectedDocuments.length > 0 && (
        <ActionBar<IDocument>
          actions={actions}
          selectedElements={selectedDocuments}
          {...actionBarProps}
        />
      )}
      <SelectableList
        mt={24}
        onSelectChange={handleSelectChange}
        selectedIndexes={selectedIndexes}
        {...selectableListProps}
      >
        {documents.map((document) => (
          <DocumentCard
            key={document.id}
            author={document.author}
            date={document.date}
            iconType={document.iconType}
            path={document.path}
            title={document.title}
          >
            {document.content}
          </DocumentCard>
        ))}
      </SelectableList>
    </div>
  );
}
