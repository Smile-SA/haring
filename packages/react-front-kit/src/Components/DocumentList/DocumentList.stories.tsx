import type { Meta, StoryObj } from '@storybook/react';

import { action } from '@storybook/addon-actions';

import { actionBarActionsMock } from '../ActionBar/ActionBar.mock';

import { DocumentList as Cmp } from './DocumentList';
import { documentsMock } from './DocumentList.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/DocumentList',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const DocumentList: IStory = {
  args: {
    actions: actionBarActionsMock,
    documents: documentsMock,
    onDocumentSelected: action('Document selected'),
    selectedDocuments: [documentsMock[1]],
  },
};
