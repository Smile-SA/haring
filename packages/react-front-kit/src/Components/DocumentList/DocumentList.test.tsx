import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { actionBarActionsMock } from '../ActionBar/ActionBar.mock';

import { DocumentList } from './DocumentList';
import { documentsMock } from './DocumentList.mock';

describe('DocumentList', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <DocumentList
        actions={actionBarActionsMock}
        documents={documentsMock}
        selectedDocuments={[documentsMock[1]]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
