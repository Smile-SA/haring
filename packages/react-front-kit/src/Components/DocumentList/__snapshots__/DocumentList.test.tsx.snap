// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentList matches snapshot 1`] = `
<div>
  <div>
    <div
      class="mantine-15edn1r"
    >
      <span>
        1 file(s) selected
      </span>
      <div
        class="mantine-Group-root mantine-19vjl6i"
      >
        <button
          class="mantine-UnstyledButton-root mantine-Button-root mantine-16xpn0"
          data-button="true"
          type="button"
        >
          <div
            class="mantine-1wpc1xj mantine-Button-inner"
          >
            <span
              class="mantine-Button-icon mantine-Button-leftIcon mantine-vuhtny"
            >
              <svg
                fill="none"
                height="16"
                stroke="currentColor"
                viewBox="0 0 12 12"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.58333 10H1.41667C1.30616 10 1.20018 9.95564 1.12204 9.87668C1.0439 9.79771 1 9.69062 1 9.57895V2.42105C1 2.30938 1.0439 2.20229 1.12204 2.12332C1.20018 2.04436 1.30616 2 1.41667 2H4.19271C4.2828 2.00037 4.37041 2.02988 4.44271 2.08421L5.89062 3.17895C5.96292 3.23327 6.05053 3.26278 6.14062 3.26316H10.5833C10.6938 3.26316 10.7998 3.30752 10.878 3.38648C10.9561 3.46544 11 3.57254 11 3.68421V5.36842"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M6 9H11"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M9 7L11 9L9 11"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
            <span
              class="mantine-Button-label mantine-yz13z6"
            >
              Move in tree
            </span>
          </div>
        </button>
        <div>
          <button
            aria-controls="mantine-f4bipx4bi-dropdown"
            aria-expanded="false"
            aria-haspopup="menu"
            class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1x9pd3r"
            id="mantine-f4bipx4bi-target"
            type="button"
          >
            <div>
              <svg
                fill="currentColor"
                height="16"
                viewBox="0 0 256 256"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M112,60a16,16,0,1,1,16,16A16,16,0,0,1,112,60Zm16,52a16,16,0,1,0,16,16A16,16,0,0,0,128,112Zm0,68a16,16,0,1,0,16,16A16,16,0,0,0,128,180Z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="mantine-Stack-root mantine-1uvvkp8"
    >
      <div
        class="mantine-Group-root mantine-1bxm023"
      >
        <div
          class="mantine-Checkbox-root mantine-yxmaw9"
        >
          <div
            class="mantine-hqx0d1 mantine-Checkbox-body"
          >
            <div
              class="mantine-13uu19v mantine-Checkbox-inner"
            >
              <input
                class="mantine-1g8epsg mantine-Checkbox-input"
                id="mantine-f4bipx4bi"
                type="checkbox"
              />
              <svg
                class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                fill="none"
                viewBox="0 0 10 7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="mantine-djumwf mantine-Checkbox-labelWrapper"
            />
          </div>
        </div>
        <div
          class="mantine-1feahhc"
        >
          <div
            class="mantine-Image-root mantine-1hgbf0p"
            style="width: 100%;"
          >
            <figure
              class="mantine-qenwvq mantine-Image-figure"
            >
              <div
                class="mantine-1iugybl mantine-Image-imageWrapper"
              >
                <img
                  class="mantine-Image-image mantine-1thijn1"
                  src=""
                  style="object-fit: cover; width: 100%; height: auto;"
                />
              </div>
            </figure>
          </div>
          <div>
            <div>
              <div
                class="mantine-mdzvt7"
              >
                <div
                  class="mantine-qu0qb1"
                >
                  <svg
                    fill="#0b7285"
                    height="38"
                    viewBox="0 0 256 256"
                    width="38"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M222,152a6,6,0,0,1-6,6H190v20h18a6,6,0,0,1,0,12H190v18a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h32A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm84,8a34,34,0,0,1-34,34H112a6,6,0,0,1-6-6V152a6,6,0,0,1,6-6h16A34,34,0,0,1,162,180Zm-12,0a22,22,0,0,0-22-22H118v44h10A22,22,0,0,0,150,180ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
                    />
                  </svg>
                </div>
                <div>
                  <div
                    class="mantine-dxs7ne"
                  >
                    <span
                      class="mantine-fakv65"
                    >
                      Random_File.PDF
                    </span>
                    <span
                      class="mantine-xrrpki"
                    >
                      (Customer &gt; 567890456 &gt; Invoices)
                    </span>
                  </div>
                  <div
                    class="mantine-1fna5yi"
                  >
                    <span>
                      Published on December 24, 2023
                    </span>
                    <span
                      class="mantine-mj4lz4"
                    >
                      -
                    </span>
                    <span
                      class="mantine-l9up18"
                    >
                      Author Name
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mantine-j5huwo"
            >
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              </p>
              <button
                class="mantine-UnstyledButton-root mantine-Button-root mantine-1tate3y"
                data-button="true"
                type="button"
              >
                <div
                  class="mantine-1wpc1xj mantine-Button-inner"
                >
                  <span
                    class="mantine-1ryt1ht mantine-Button-label"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 256 256"
                      width="12"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224,152v56a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V152a8,8,0,0,1,16,0v56H208V152a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,132.69V40a8,8,0,0,0-16,0v92.69L93.66,106.34a8,8,0,0,0-11.32,11.32Z"
                      />
                    </svg>
                    <div
                      class="mantine-151bw52"
                    />
                    PDF, FR - 1Mo
                  </span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mantine-Group-root mantine-1bxm023"
      >
        <div
          class="mantine-Checkbox-root mantine-yxmaw9"
        >
          <div
            class="mantine-hqx0d1 mantine-Checkbox-body"
          >
            <div
              class="mantine-13uu19v mantine-Checkbox-inner"
            >
              <input
                checked=""
                class="mantine-1g8epsg mantine-Checkbox-input"
                id="mantine-f4bipx4bi"
                type="checkbox"
              />
              <svg
                class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                fill="none"
                viewBox="0 0 10 7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="mantine-djumwf mantine-Checkbox-labelWrapper"
            />
          </div>
        </div>
        <div
          class="mantine-1feahhc"
        >
          <div
            class="mantine-Image-root mantine-1hgbf0p"
            style="width: 100%;"
          >
            <figure
              class="mantine-qenwvq mantine-Image-figure"
            >
              <div
                class="mantine-1iugybl mantine-Image-imageWrapper"
              >
                <img
                  class="mantine-Image-image mantine-1thijn1"
                  src=""
                  style="object-fit: cover; width: 100%; height: auto;"
                />
              </div>
            </figure>
          </div>
          <div>
            <div>
              <div
                class="mantine-mdzvt7"
              >
                <div
                  class="mantine-qu0qb1"
                >
                  <svg
                    fill="#0b7285"
                    height="38"
                    viewBox="0 0 256 256"
                    width="38"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M222,152a6,6,0,0,1-6,6H202v50a6,6,0,0,1-12,0V158H176a6,6,0,0,1,0-12h40A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm80,0a26,26,0,0,1-26,26H122v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h16A26,26,0,0,1,158,172Zm-12,0a14,14,0,0,0-14-14H122v28h10A14,14,0,0,0,146,172ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
                    />
                  </svg>
                </div>
                <div>
                  <div
                    class="mantine-dxs7ne"
                  >
                    <span
                      class="mantine-fakv65"
                    >
                      Presentation.PPT
                    </span>
                    <span
                      class="mantine-xrrpki"
                    >
                      (Customer &gt; 567890456 &gt; Invoices)
                    </span>
                  </div>
                  <div
                    class="mantine-1fna5yi"
                  >
                    <span>
                      Published on December 24, 2023
                    </span>
                    <span
                      class="mantine-mj4lz4"
                    >
                      -
                    </span>
                    <span
                      class="mantine-l9up18"
                    >
                      Author Name
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mantine-j5huwo"
            >
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              </p>
              <button
                class="mantine-UnstyledButton-root mantine-Button-root mantine-1tate3y"
                data-button="true"
                type="button"
              >
                <div
                  class="mantine-1wpc1xj mantine-Button-inner"
                >
                  <span
                    class="mantine-1ryt1ht mantine-Button-label"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 256 256"
                      width="12"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224,152v56a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V152a8,8,0,0,1,16,0v56H208V152a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,132.69V40a8,8,0,0,0-16,0v92.69L93.66,106.34a8,8,0,0,0-11.32,11.32Z"
                      />
                    </svg>
                    <div
                      class="mantine-151bw52"
                    />
                    PPT, FR - 1Mo
                  </span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mantine-Group-root mantine-1bxm023"
      >
        <div
          class="mantine-Checkbox-root mantine-yxmaw9"
        >
          <div
            class="mantine-hqx0d1 mantine-Checkbox-body"
          >
            <div
              class="mantine-13uu19v mantine-Checkbox-inner"
            >
              <input
                class="mantine-1g8epsg mantine-Checkbox-input"
                id="mantine-f4bipx4bi"
                type="checkbox"
              />
              <svg
                class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                fill="none"
                viewBox="0 0 10 7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="mantine-djumwf mantine-Checkbox-labelWrapper"
            />
          </div>
        </div>
        <div
          class="mantine-1feahhc"
        >
          <div
            class="mantine-Image-root mantine-1hgbf0p"
            style="width: 100%;"
          >
            <figure
              class="mantine-qenwvq mantine-Image-figure"
            >
              <div
                class="mantine-1iugybl mantine-Image-imageWrapper"
              >
                <img
                  class="mantine-Image-image mantine-1thijn1"
                  src=""
                  style="object-fit: cover; width: 100%; height: auto;"
                />
              </div>
            </figure>
          </div>
          <div>
            <div>
              <div
                class="mantine-mdzvt7"
              >
                <div
                  class="mantine-qu0qb1"
                >
                  <svg
                    fill="#0b7285"
                    height="38"
                    viewBox="0 0 256 256"
                    width="38"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M222,152a6,6,0,0,1-6,6H190v20h18a6,6,0,0,1,0,12H190v18a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h32A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm84,8a34,34,0,0,1-34,34H112a6,6,0,0,1-6-6V152a6,6,0,0,1,6-6h16A34,34,0,0,1,162,180Zm-12,0a22,22,0,0,0-22-22H118v44h10A22,22,0,0,0,150,180ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
                    />
                  </svg>
                </div>
                <div>
                  <div
                    class="mantine-dxs7ne"
                  >
                    <span
                      class="mantine-fakv65"
                    >
                      Other_random_File.PDF
                    </span>
                    <span
                      class="mantine-xrrpki"
                    >
                      (Customer &gt; 567890456 &gt; Invoices)
                    </span>
                  </div>
                  <div
                    class="mantine-1fna5yi"
                  >
                    <span>
                      Published on December 24, 2023
                    </span>
                    <span
                      class="mantine-mj4lz4"
                    >
                      -
                    </span>
                    <span
                      class="mantine-l9up18"
                    >
                      Author Name
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mantine-j5huwo"
            >
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
              </p>
              <button
                class="mantine-UnstyledButton-root mantine-Button-root mantine-1tate3y"
                data-button="true"
                type="button"
              >
                <div
                  class="mantine-1wpc1xj mantine-Button-inner"
                >
                  <span
                    class="mantine-1ryt1ht mantine-Button-label"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 256 256"
                      width="12"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224,152v56a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V152a8,8,0,0,1,16,0v56H208V152a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,132.69V40a8,8,0,0,0-16,0v92.69L93.66,106.34a8,8,0,0,0-11.32,11.32Z"
                      />
                    </svg>
                    <div
                      class="mantine-151bw52"
                    />
                    PDF, FR - 1Mo
                  </span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
